pkg/oauth/apiserver/auth.go

```diff
@@ -156,13 +127,24 @@ func (c *OAuthServerConfig) WithOAuth(handler http.Handler) (http.Handler, error
 		osinserver.NewDefaultErrorHandler(),
 	)
 	server.Install(mux, oauthutil.OpenShiftOAuthAPIPrefix)
+	c.oauthServerMux = mux
 
-	if err := CreateOrUpdateDefaultOAuthClients(c.Options.MasterPublicURL, c.AssetPublicAddresses, clientRegistry); err != nil {
-		glog.Fatal(err)
+	// glog.Infof("oauth server configured as: %#v", server)
+	// glog.Infof("auth handler: %#v", authHandler)
+	// glog.Infof("auth request handler: %#v", authRequestHandler)
+	// glog.Infof("grant checker: %#v", grantChecker)
+	// glog.Infof("grant handler: %#v", grantHandler)
+
+	return baseMux, nil
+}
+
+func (c *OAuthServerConfig) RegisterTokenRequestEndpoint(context genericapiserver.PostStartHookContext) error {
```

Comment:
This bit is nasty. The server won't report healthy until it succeeds though.  We need to decide where this should logically live, but I *think* this is the part where we actually depend on it.